utils/openocd_cmd: add support for all available open nodes

aabadie opened the pull request:
This PR allows to easily flash any kind of open nodes that support openocd from a gateway.

aabadie committed:
utils/openocd_cmd: add support for all available open nodes

rienafairefr commented:
You could use the node plugin registry instead of hardcoding the nodes? Then you can check if the class is subtype of `NodeOpenOcd`.

aabadie commented:
> You could use the node plugin registry instead of hardcoding the nodes? Then you can check if the class is subtype of NodeOpenOcd.

Thinking of it and I like the idea. I'll give it a try.

aabadie committed:
utils/openocd_cmd: generalize logic by using registered nodes

aabadie committed:
utils/avrdude_cmd: generalize logic by using registered nodes

aabadie committed:
nodes: remove useless try/except

aabadie commented:
@rienafairefr, I updated the PR like you suggested also applied the same logic for avrdude based nodes.
I tested with an M3 and ST Lrwan1 board and it works fine. Even without the try/except in the automatic nodes module importer logic, so I also removed it!

rienafairefr commented on gateway_code/nodes.py:
```diff
@@ -116,10 +116,7 @@ def import_all_nodes(pkg_dir):
-            module_loader.find_module(name).load_module(name)
-        except ImportError:
-            pass
+        module_loader.find_module(name).load_module(name)
```
weird, but good, that it now works :-)

rienafairefr commented:
Did you test on an avr based board as well ?

aabadie commented:
> Did you test on an avr based board as well ?

No. It's possible on one of the CI node (there's a zigduino on Strasbourg and a Leonardo in Grenoble).

aabadie commented:
@rienafairefr, I tested on leonardo and zigduino and it also works. I think we can merge :)
